src/ray/raylet/worker_pool.cc

```diff
@@ -765,7 +765,7 @@ void WorkerPool::OnWorkerStarted(const std::shared_ptr<WorkerInterface> &worker)
   // This is a workaround to finish driver registration after all initial workers are
   // registered to Raylet if and only if Raylet is started by a Python driver and the
   // job config is not set in `ray.init(...)`.
-  if (first_job_ == worker->GetAssignedJobId() &&
+  if (worker->GetAssignedJobId().IsNil() && worker_type == rpc::WorkerType::WORKER &&
```

Comment:
(I'm guessing not because of [this change](https://github.com/ray-project/ray/pull/30883/files#diff-4c19d6ebd8575189e9439fa5646b567546dc54f64d576a959ce1bb43902d1a03R806-R808), but wondering if there is any other edge case)

Reply:
good point... hmm

Reply:
i think a safer check is just to check if sufficient number of workers started, ignore the job_id.